File: 383.Ransom_Note/main.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

class Solution
{
public:
     bool canConstruct(string ransomNote, string magazine)
     {
          vector<int> freq(26, 0);
          for (char c : magazine)
          {
               freq[c - 'a']++;
          }
          for (char c : ransomNote)
          {
               if (--freq[c - 'a'] < 0)
               {
                    return false;
               }
          }
          return true;
     }
};

int main()
{
     Solution solution;
     string ransomNote = "aa";
     string magazine = "aab";
     cout << (solution.canConstruct(ransomNote, magazine) ? "true" : "false") << endl; // Output: true
     return 0;
}

File: 383.Ransom_Note/main.py
class Solution:
    def can_construct(self, ransom_note: str, magazine: str) -> bool:
        freq = [0] * 26
        for char in magazine:
            freq[ord(char) - ord('a')] += 1
        for char in ransom_note:
            if freq[ord(char) - ord('a')] == 0:
                return False
            freq[ord(char) - ord('a')] -= 1
        return True

if __name__ == "__main__":
    solution = Solution()
    ransom_note = "aa"
    magazine = "aab"
    print(solution.can_construct(ransom_note, magazine))  # Output: true
